pkg/apis/openebs.io/scp/v1alpha1/storage-pool.go

```diff
@@ -29,8 +31,387 @@ import (
 type StoragePool struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec is the spec for a StoragePool resource
+	Spec StoragePoolSpec `json:"spec"`
+	// Status is for handling status of StoragePool resource
+	Status StoragePoolStatus `json:"status,omitempty"`
 }
 
+// StoragePoolSpec is the spec for a StoragePool resource
+type StoragePoolSpec struct {
+	// StorageCohortID of the cohort the pool is a part of
+	StorageCohortID string `json:"storageCohortID,omitempty"`
+
+	// Type of the storage pool, viz lvm, spdk blobstor
+	Type StoragePoolType `json:"type,omitempty"`
```

Comment:
Thinking if we need it ...
We can determine pool type from pool configuration..

Reply:
Will keep it as of now, let me know if this needs to be removed later.